ReactWindows/ReactNative/Touch/TouchHandler.cs

```diff
@@ -84,7 +84,9 @@ private void OnPointerPressed(object sender, PointerRoutedEventArgs e)
 
             var originalSource = e.OriginalSource as DependencyObject;
             var rootPoint = e.GetCurrentPoint(_view);
-            var reactView = GetReactViewTarget(originalSource, rootPoint.Position);
+            var transform = ((UIElement)sender).TransformToVisual(Window.Current.Content);
+            Point p = transform.TransformPoint(rootPoint.Position);
+            var reactView = GetReactViewTarget(originalSource, p);
```

Comment:
Should we add a check in TouchHandler to see if we even need to do this? We could do something in the constructor that checked if the current window content was a root view, and only if it's not, transform twice.

Reply:
That way apps that are only react-native only pay the penalty of a small equality check.

---
In reply to: [172059663](https://github.com/Microsoft/react-native-windows/pull/1556#discussion_r172059663) [](ancestors = 172059663)

Reply:
Currently, the TouchHandler is constructed prior to Window.Current.Content being assigned to (the Page that contains) the ReactRootView.  I could defer that to a Loaded event handler.  i.e. :

        public ReactRootView()
        {
            this.Loaded += OnLoad;
        }

        private void OnLoad(object sender, RoutedEventArgs e)
        {
            TouchHandler = new TouchHandler(this);
            this.Loaded -= OnLoad;
        }

But, this would seem to encode the assumption that it's in a page.  i.e. :

	_isRootedInWindow = (view.Parent.GetType() == typeof(Page)) && (((Page)view.Parent).Parent == Window.Current.Content);

IMHO, it's simpler/safer to do the transform each time.  Let me know what you think...

Reply:
We could also check once then cache the value, although, in any case, it's not guaranteed that the ReactRootView wouldn't become the Window.Current.Content at any point in time, or for that matter that we wouldn't replace the root view in Window.Current.Content and re-attach it as a subview. You're probably right that checking every time is safer. It's only for PointerPressed events at any rate.

Reply:
Are you OK with leaving this as-is, then?

Reply:
yes.

Reply:
I believe I'm blocked from merging due to this requested change.  Could you clear it out?  Thanks.